src/layers/conv.jl

```diff
@@ -223,6 +221,60 @@ function Base.show(io::IO, l::ChebConv)
     print(io, ")")
 end
 
+struct DotProductAttention{W <: AbstractArray} <: GNNLayer
+    channel::Pair{Int, Int}
+    Wq::W
+    Wk::W
+    Wv::W
+end
+
+function DotProductAttention(ch::Pair{Int, Int}, init = glorot_uniform)
+    chin, chout = ch
+    Wq = init(chin, chin)
+    Wk = init(chin, chin)
+    Wv = init(chout, chin)
+    DotProductAttention(ch, Wq, Wk, Wv)
+end
+
+function (l::DotProductAttention)(g::AbstractGNNGraph, x)
+    q = l.Wq * x
+    k = l.Wk * x
+    v = l.Wv * x
+    x, _ = dot_product_attention(q, k, v)
+    return x
+end
+@functor DotProductAttention
+
+@doc raw"""
+    GPSConv(in=>out, conv=GCNConv, σ=identity; bias=true, heads=5, init=glorot_uniform)
+"""
+struct GPSConv <: GNNLayer
+    convlayer::GNNChain
+    attnlayer::GNNChain
+    ffn::GNNChain
+end
+
+function GPSConv(ch::Pair{Int, Int}, gconv, σ = identity; init = glorot_uniform)
+    in, out = ch
+    gattn = DotProductAttention(ch)
+    convlayer = GNNChain(gconv, Dropout(0.5), LayerNorm(out))
+    attnlayer = GNNChain(gattn, Dropout(0.5), LayerNorm(out))
+    ffn = GNNChain(Dense(in => 2 * in, σ), Dropout(0.5), Dense(2 * in => in), Dropout(0.5))
+    return GPSConv(convlayer, attnlayer, ffn)
+end
+
+function (l::GPSConv)(g::AbstractGNNGraph, x)
+    check_num_nodes(g, x)
+    f = GNNChain(
+            Parallel(+,
+                Parallel(+, l.convlayer, identity),
+                Parallel(+, l.attnlayer, identity),
+            ),
+            l.ffn
+    )
```

Comment:
We should avoid constructing a gnnchain in each forward pass. Also, the use of parallel makes the sequence of operations hard to parse. 
We should just have a plain sequence of transformations of the input here, like:
```julia
x1 = l.bn1(l.dropout1(l.conv(g, x)) .+ x))
x2 =l.bn2(l.dropout2(l.attn(x)) .+ x))
y = l.mlp( x1 + x2) # not sure if we should also add a skip connection here
```
Notice the order of operations. It is different from what the layer is currently doing. Current implementation doesn't seem to follow eqs.  (6)-(11) in the paper.